client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +642,88 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_presets_by_filtering(export_preset_name, channel_type_list):
+    """Function to get export presets included with specific channels
+    requested by users.
+
+    Args:
+        export_preset_name (str): Name of export preset
+        channel_type_list (list): A list of channel type requested by users
+
+    Returns:
+        dict: export preset data
+    """
+
+    new_maps = []
+
+    export_presets = get_export_presets()
+    export_preset_nice_name = export_presets[export_preset_name]
+    resource_presets = substance_painter.export.list_resource_export_presets()
+    preset = next(
+        (
+            preset for preset in resource_presets
+            if preset.resource_id.name == export_preset_nice_name
+        ), None
+    )
+    if preset is None:
+        return {}
+    maps = preset.list_output_maps()
+    for channel_map in maps:
+        for n in channel_type_list:
+            if not channel_map.get("fileName"):
+                continue
+
+            if n in channel_map["fileName"]:
+                new_maps.append(channel_map)
+    # Create a new preset
+    return {
+        "exportPresets": [
+            {
+                "name": export_preset_name,
+                "maps": new_maps
+            }
+        ],
+    }
+
+
+@contextlib.contextmanager
+def set_layer_stack_opacity(node_ids, channel_type):
+    """Function to set the opacity of the layer stack during
+    context
+    Args:
+        node_ids (list): A list of substance painter node ids
+        channel_type (list): A list of channel types
+    """
+
+    all_selected_nodes = []
+    opacity_set_list = []
+    stack = substance_painter.textureset.get_active_stack()
+    stack_root_layers = (
+        substance_painter.layerstack.get_root_layer_nodes(stack)
+    )
+    # Do nothing
+    if not node_ids or not channel_type:
+        yield
+        return
```

Comment:
No need to perform any of these calls if we're not doing anything anyway?
```suggestion
    # Do nothing
    if not node_ids or not channel_type:
        yield
        return

    all_selected_nodes = []
    opacity_set_list = []
    stack = substance_painter.textureset.get_active_stack()
    stack_root_layers = (
        substance_painter.layerstack.get_root_layer_nodes(stack)
    )
```